lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -1273,6 +1274,15 @@ bool TypeLoweringVisitor::visitExpr(RefResolveOp op) {
   return lowerProducer(op, clone, op.getRef().getType());
 }
 
+bool TypeLoweringVisitor::visitExpr(RefCastOp op) {
+  auto clone = [&](const FlatBundleFieldEntry &field,
+                   ArrayAttr attrs) -> Value {
+    auto input = getSubWhatever(op.getInput(), field.index);
+    return builder->create<RefCastOp>(RefType::get(field.type), input);
```

Comment:
ref.sub demotes rwprobe to probe, but since we don't break up rwprobe's that's not a concern here :+1: .